R/shoji-catalog.R

```diff
@@ -16,8 +16,28 @@ getListSlot <- function (x, i, what=character(1), ifnot=NA_character_) {
 }
 
 setIndexSlot <- function (x, i, value, unique=FALSE) {
-    if (length(value) == 1) value <- rep(value, length(x))
-    stopifnot(length(x) == length(value))
+    scalar_value <- length(value) == 1
+    if (!scalar_value) {
+        stopifnot(length(x) == length(value))
+    }
+
+    duples <- duplicated(urls(x))
+    if (any(duples)) {
+        ## This is probably a mistake, a selection from a catalog that wasn't
+        ## unique. Make sure that the user isn't trying to assign different
+        ## values for the duplicated entries, and if not, clean it up
+        if (!scalar_value && !identical(duples, duplicated(paste(urls(x), value)))) {
+            halt("Can't update an index with duplicated entries")
```

Comment:
I'm not sure that this error will be clear what to do to solve it. What about something like "Can't update the same index item with more than one entry" I really wish we could abstract away index and entry and talk about names/aliases but this could get called in other circumstances.

Reply:
Yeah I kinda punted on the error message, partly because it was hard to trigger. If you can think of better words, cool. Either way, i don't expect this error to get a lot of visibility. It's more a backstop to intercept what would be an even more opaque 400 Bad Request from the server.